app/controllers/sections/dispatch/DispatchUseConsignorDetailsController.scala

```diff
@@ -53,26 +54,21 @@ class DispatchUseConsignorDetailsController @Inject()(override val messagesApi:
     authorisedDataRequestAsync(ern, draftId) { implicit request =>
       formProvider().bindFromRequest().fold(
         renderView(BadRequest, _, mode),
-        value => {
-          if (request.userAnswers.get(DispatchUseConsignorDetailsPage).contains(value)) {
-            Future(Redirect(navigator.nextPage(DispatchUseConsignorDetailsPage, mode, request.userAnswers)))
-          } else {
-
-            val cleanedUserAnswers = if (!value) request.userAnswers else request.userAnswers
-              .remove(DispatchBusinessNamePage)
-              .remove(DispatchAddressPage)
-
-            saveAndRedirect(
-              page = DispatchUseConsignorDetailsPage,
-              answer = value,
-              currentAnswers = cleanedUserAnswers,
-              mode = NormalMode
-            )
-          }
-        }
+        handleAnswerAndRedirect
       )
     }
 
   private def renderView(status: Status, form: Form[Boolean], mode: Mode)(implicit request: DataRequest[_]): Future[Result] =
     Future.successful(status(view(form, routes.DispatchUseConsignorDetailsController.onSubmit(request.ern, request.draftId, mode))))
+
+  private def handleAnswerAndRedirect(value: Boolean)(implicit request: DataRequest[_]): Future[Result] = {
+    val updatedAnswers = if(request.userAnswers.get(DispatchUseConsignorDetailsPage).contains(value)) request.userAnswers else {
+      ((value, request.userAnswers.get(ConsignorAddressPage)) match {
+        case (true, Some(address)) => request.userAnswers.set(DispatchAddressPage, address)
```

Comment:
Theoretically yes; the only way around this would be to use the Consignor Address to pre-populate and only save on submission. That is probably a better solution, so I'll refactor this. Just trying to think if there's any consequences to that, but I don't think there is.